src/modules/welcome/checker/RequirementsChecker.cpp

```diff
@@ -337,8 +339,20 @@ RequirementsChecker::checkHasPower()
 bool
 RequirementsChecker::checkHasInternet()
 {
-    // default to true in the QNetworkAccessManager::UnknownAccessibility case
-    bool hasInternet = QNetworkAccessManager(this).networkAccessible() != QNetworkAccessManager::NotAccessible;
+    // doc from: https://doc.qt.io/qt-5/qnetworkconfigurationmanager.html#isOnline
+    bool hasInternet = false;
+
+    QNetworkConfigurationManager mgr;
+    QList<QNetworkConfiguration> activeConfigs = mgr.allConfigurations(QNetworkConfiguration::Active);
+
+    if (activeConfigs.count() > 0) {
+        Q_ASSERT(mgr.isOnline());
+        hasInternet = mgr.isOnline();
```

Comment:
@AlmAck I've just pushed a change to master which should improve the situation. Please test and let me know if it fixes your use case.